componentLibraries/defaultLibrary/Special/AeroComponents/AeroFlightRecorder.hpp

```diff
@@ -0,0 +1,339 @@
+/*-----------------------------------------------------------------------------
+
+ Copyright 2017 Hopsan Group
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+
+ The full license is available in the file LICENSE.
+ For details about the 'Hopsan Group' or information about Authors and
+ Contributors see the HOPSANGROUP and AUTHORS files that are located in
+ the Hopsan source code root directory.
+
+-----------------------------------------------------------------------------*/
+
+//!
+//! @file   AeroFlightRecorder.hpp
+//! @author Alejandro Sobron <alejandro.sobron@liu.se>
+//! @date   2020-05-26
+//!
+//! @brief Contains a log-file generator component for geographical visualization
+//! @ingroup AeroComponents
+//!
+
+#ifndef AEROFLIGHTRECORDER_HPP_INCLUDED
+#define AEROFLIGHTRECORDER_HPP_INCLUDED
+
+#include "ComponentEssentials.h"
+#include "ComponentSystem.h"
+#include "ComponentUtilities.h"
+#include "math.h"
+#include <iostream>
+#include <fstream>
+#include <ctime>
+
+namespace hopsan {
+
+    class AeroFlightRecorder : public ComponentSignal
+    {
+
+    private:
+    // Private member variables
+
+        //inputVariables pointers
+        double *mpLongitude;
+        double *mpLatitude;
+        double *mpAltitude;
+        double *mpPhi;
+        double *mpTheta;
+        double *mpPsi;
+        double *mpV;
+
+        //inputParameters pointers
+        double *mpDt;
+        double *mpScaleX;
+        double *mpScaleY;
+        double *mpScaleZ;
+
+        //inputVariables
+        double longitude;
```

Comment:
I know that many of the other components put all of their variables here in the private part like this.
But the problem is that all of them become member variables. Even those that are only used locally inside the individual functions. Any new component should not repeat this old mistake. 

So can you please move all of the (non mName) members down to their respective functions.
That way the compiler can better optimize the code, and does not need to allocate persistent memory for these variables.
Also the variable dt is not being used.

Reply:
You are absolutely right in here. I just followed the "style" I saw in older components without even thinking twice about this issue. I'll fix it right away!